cadence/contracts/Giefts.cdc

```diff
@@ -16,23 +17,199 @@ pub contract Giefts {
     //                            PATHS                            //
     /////////////////////////////////////////////////////////////**/
 
+    pub let GieftsStoragePath: StoragePath
+    pub let GieftsPublicPath: PublicPath
+    pub let GieftsPrivatePath: PrivatePath
 
     /**//////////////////////////////////////////////////////////////
     //                            EVENTS                           //
     /////////////////////////////////////////////////////////////**/
 
+    pub event Packed(gieft: UInt64, nfts: [UInt64])
+    pub event Added(gieft: UInt64, nft: UInt64, type: String, name: String, thumbnail: String)
+    pub event Removed(gieft: UInt64, nft: UInt64, type: String, name: String, thumbnail: String)
+    pub event Claimed(gieft: UInt64, nft: UInt64)
```

Comment:
sorry I was not clear here, claimed is the event that I think is needed to be human readable. And come to think of it adding sender/receiver or giver/taker addresses here is good too. So i think we need a way to expose CollectionData from GieftPublic so that we can create the collection first and then send it into claimNFT